llama_index/storage/docstore/keyval_docstore.py

```diff
@@ -150,6 +150,91 @@ def add_documents(
                         node_key, metadata, collection=self._metadata_collection
                     )
 
+    async def aadd_documents(
```

Comment:
spicy take maybe, but I think I would prefer `async_add_documents` -- this follows a similar pattern in our vector stores to avoid the double `aa`

Reply:
also, I wonder if there is a way to refactor this function slightly so that we aren't maintaining two copies of the function. 

This one in particular is quite complex, so I'd like to maintain a more singular implementation. I'll take a few minutes to see if I have an idea for this

Reply:
@logan-markewich haha yeah its definitely spicy. Typing the double a felt dirty. The `async_add_documents` is way better.